src/controllers/authenticationController.ts

```diff
@@ -170,6 +173,41 @@ export default class AuthenticationController {
 
     const spidUser = errorOrSpidUser.right;
 
+    // check for CIE IDP test login
+    const maybeCieTestUserInWhitelist = pipe(
+      Object.keys(CIE_IDP_IDENTIFIERS).filter(
+        (identifier) =>
+          identifier !==
+          "https://idserver.servizicie.interno.gov.it/idp/profile/SAML2/POST/SSO"
+      ),
+      // if issuer is one of CIE test identifiers
+      O.fromPredicate(ROA.some((identifier) => spidUser.issuer === identifier)),
+      // and if the spidUser fiscalcode is in whitelist
+      O.fold(
+        // some here to let the code continue the normal logic flow
+        // in case the issuer is not a CIE_TESTING one
```

Comment:
Modifying the url value will not cause the build to fail.
We could refactor this way: 


```typescript 
    const cieProduzioneIdentifier =
      "https://idserver.servizicie.interno.gov.it/idp/profile/SAML2/POST/SSO";

    // Define test identifiers as everything by production url
    type CIETestIdentifiers = keyof Omit<
      typeof CIE_IDP_IDENTIFIERS,
     // Pick is needed to ensure 'cieProduzioneIdentifier' value is contained in CIE_IDP_IDENTIFIERS
      keyof Pick<typeof CIE_IDP_IDENTIFIERS, typeof cieProduzioneIdentifier>
    >;

   // Define a predicate for checking is user is a CIE test user
    const isCIETestUser = (
      u: SpidUser
    ): u is SpidUser & { readonly issuer: CIETestIdentifiers } =>
      u.issuer !== cieProduzioneIdentifier;

    // check for CIE IDP test login
    const maybeCieTestUserInWhitelist = pipe(
      spidUser,
      // if issuer is one of CIE test identifiers
      O.fromPredicate(isCIETestUser),
      // and if the spidUser fiscalcode is in whitelist
      O.fold(
        // some here to let the code continue the normal logic flow
        // in case the issuer is not a CIE_TESTING one
      ```